### PR TITLE
[WIP][FIX] Adds apply_baseline to Evoked

### PR DESCRIPTION
**Pull request by gamazeps**
Also mentions that the 'baseline' parameter will be removed from the
constructor.

Note: there is no test for the moment, nor additions to whats_new.rst

Fixes issue #3268

### CODE REVIEW

File: mne/evoked.py [hunk 1]
```diff
@@ -115,6 +117,37 @@ def __init__(self, fname, condition=None, baseline=None, proj=True,
             self.apply_proj()
         self.data = rescale(self.data, self.times, baseline, copy=False)
 
+    @verbose
+    def apply_baseline(self, baseline, verbose=None):
+        """Baseline correct evoked data
+
+        Parameters
+        ----------
+        baseline : tuple of length 2
```
Comment (Eric89GXL) [inline mne/evoked.py]: This could probably default to `(None, 0)`, same with `Epochs.apply_baseline`. The latter is backward-compatible change so we can do it. That would make them consistent with `Epochs.__init__`. But we should wait to see if others agree.

Comment (agramfort) [inline mne/evoked.py]: +1 to default to (None, 0)

**Comment by agramfort** [comment]
@gamazeps you'll need to add a test and update the rest of the code to make sure the deprecation warning is not thrown in any test / example

File: mne/evoked.py (hunk 1 above)
Comment (gamazeps) [inline mne/evoked.py]: Sounds rather logic, but shouldn't that be in another commit (as it can be seen as orthogonal to this one) ?

Comment (agramfort) [inline mne/evoked.py]: PR == many commits

so push your commits in the branch to append the commits. We'll merge when
it's complete

Comment (gamazeps) [inline mne/evoked.py]: Since read_evokeds also has baseline as a parameter, should it be deprecated there too or should we use the `apply_baseline` inside it ?

Comment (Eric89GXL) [inline mne/evoked.py]: I guess we can just use `apply_baseline` inside it

**Comment by gamazeps** [comment]
@agramfort Evoked is actually never created outside of `read_evokeds` so this is nice (and fixed).

The bad thing is that the use of the baseline correction was never tested before on Evoked (only on Epoch) and I must confess that I wouldn't feel comfortable doing that myself (I could, but I'm quite likely to mess it up). I could nevertheless try or open an issue for that, what's your opinion ?

**Comment by Eric89GXL** [comment]
Are you up for a couple of iterations? You can follow what is already in `mne/tests/test_evoked.py`, create something like:
```
def test_evoked_baseline:
    """Test evoked baseline"""
    evoked = mne.read_evokeds(..., baseline=None)
```
And then 1) do some operations on `evoked` that you would expect to change the baseline, and 2) test that those things actually happened e.g. with `np.allclose` or so. If you don't have time, I can take over your PR and do it, but hopefully it won't be too bad.

**Comment by gamazeps** [comment]
If you are up for correcting my potential mistakes, i'm up for it :)

**Comment by Eric89GXL** [comment]
Sure go ahead, there are plenty of examples in `mne/tests/test_evoked.py` that you can work off of. Actually there should be material you can take with little modification required from `mne/tests/test_epochs.py` since we should already have tests for `Epochs.apply_baseline` in there.

**Comment by dengemann** [comment]
This looks clean to me, I like this idea.

**Comment by gamazeps** [comment]
@Eric89GXL sorry if it took me a while to do the commit.

Should we also open the baseline correction `mode` in `apply_baseline` as it will always do a `mean` baseline correction even though more are possible ?

I propose adding the argument to both Evoked and Epoch and have it default to `mean`, does that sound good to you ?

**Comment by agramfort** [comment]
forget about the mode here. it's YAGNI for such objects AFAIK

**Comment by gamazeps** [comment]
I also have one small issue with the testing.

The logging testing uses the logs of the baseline correction, the issue here is that the the `baseline.rescale` is called twice, once in the constructor of `Evoked` (as I assumed we did not want to remove it) and once in `read_evoked` (where the `apply baseline` method is called), thus the log contain the logs generated by `baseline.rescale` twice.

Should I simply make the baseline argument useless in `Evoked.__init__` or add a line in the test logs ?

Also why are [these lines](https://github.com/mne-tools/mne-python/blob/master/mne/tests/test_utils.py#L259-L264) there instead of changing the test logs (I assume that there is a good reason but I don't quite get it).

**Comment by Eric89GXL** [comment]
> Also why are these lines there instead of changing the test logs (I assume that there is a good reason but I don't quite get it).

I can't remember why I did it that way, but it probably isn't necessary

LGTM +1 for merge

**Comment by Eric89GXL** [comment]
(assuming you are done, that is -- if so please set the title beginning to MRG)

**Comment by dengemann** [comment]
Thank you @gamazeps !

**Comment by gamazeps** [comment]
@dengemann one test is [not passing](https://travis-ci.org/mne-tools/mne-python/jobs/142245705) though, you may want to revert the commits  :/

**Comment by dengemann** [comment]
@gamazeps ah damnit, I was too fast. But it seems it's a simple pep8 error, I'll fix it on master.

**Comment by gamazeps** [comment]
@dengemann just pushed it to this branch

**Comment by dengemann** [comment]
already taken care of. thanks.
